container/lxd/network.go

```diff
@@ -0,0 +1,181 @@
+package lxd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/juju/errors"
+	"github.com/lxc/lxd/shared/api"
+
+	"github.com/juju/juju/network"
+)
+
+// LocalBridgeName returns the name of the local LXD network bridge.
+func (s *Client) LocalBridgeName() string {
+	return s.localBridgeName
+}
+
+func (s *Client) VerifyDefaultBridge(profile *api.Profile) error {
+	eth0, ok := profile.Devices["eth0"]
```

Comment:
anyplace where we are hard coding the string "eth0" is *definitely* suspect.
We do it in places for LXD but we really really shouldn't be.
We should at least have a bug/tracking the places where we are doing this.

Side note: you're probably fixing https://bugs.launchpad.net/juju/+bug/1668547

Reply:
I image we should be looking for *any* device that has the bridge as a parent.
I think the new commit should address that bug. Unit and system tests indicate so.